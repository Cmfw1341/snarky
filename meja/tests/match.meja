let x = fun (y : int, z : bool) => {
  switch (y) {
    | i => z
  };
};

let y = fun (f : int -> int -> int -> _) => {
  switch (1i, 2i, 3i) {
    | (i, j, k) => f (i, j, k)
  };
};

type t = {a: int, b: bool};

type u = {f : int -> int};

type v('a, 'b) = {x : 'a, y : 'b, g : 'a -> 'b};

let z = fun (x, {f}) => {
  switch (x) {
    | {a: x, b} => f (x)
  };
};

let a = fun (x : t, {f}) => {
  switch (x) {
    | {a: x, b} => f (x)
  };
};

let b = switch ({f: fun (x) => { 12i; }}) {
  | {f} => f
};

let c : _ -> v (int, bool) = fun (x : v ('a, 'b)) => {
  switch (x) {
    | {x, y, g: f} => {x: 12i, y, g: fun (_) => { f (x); }}
  };
};

type w('a) = A | B(int, int) : w(int) | C (w('b)) : w('b);

let d = fun (x) => {
  switch (x) {
    | A => 1i
    | B(x, y) => y
    | C(A) => 2i
    | C(B(x, y)) => y
    | _ => 3i
  };
};

let if_ = fun (x : bool, y : 'a, z : 'a) => {
  switch (x) {
    | true => y
    | false => z
  };
}
