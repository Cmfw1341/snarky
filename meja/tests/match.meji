let x : int -> bool -> bool;

let y : (int -> int -> int -> _) -> _;

type t = {a: int, b: bool};

type u = {f: int -> int};

convertible type v('a, 'b) = {x: 'a, y: 'b, g: 'a -> 'b}
with prover v('a, 'b) = {x: 'a, y: 'b, g: 'a -> 'b}
by v_typ;

let z : t -> u -> int;

let a : t -> u -> int;

let b : int -> int;

let c : v('a, bool) -> v(int, bool);

type w('a) = A | B(int, int) : w(int) | C(w('b)) : w('b);

let d : w(int) -> int;

let if_ : bool -> 'a -> 'a -> 'a;


