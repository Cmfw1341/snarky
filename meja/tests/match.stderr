File "tests/match.meja", line 55, characters 6-10:
Error: Unbound constructor true.
